versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTableBackend.java

```diff
@@ -192,43 +193,35 @@ private void eraseRepositoriesNoAdminClient(Collection<String> repositoryIds) {
   }
 
   private void eraseRepositoriesTable(String tableId, List<ByteString> prefixes) {
-    BulkMutation bulkDelete = BulkMutation.create(tableId);
 
-    Filters.InterleaveFilter repoFilter = FILTERS.interleave();
-    for (ByteString prefix : prefixes) {
-      repoFilter.filter(FILTERS.key().regex(prefix.concat(REPO_REGEX_SUFFIX)));
-    }
-
-    Query.QueryPaginator paginator = Query.create(tableId).filter(repoFilter).createPaginator(100);
-    Iterator<Row> rows = dataClient.readRows(nextQuery(paginator)).iterator();
-    while (true) {
-      ByteString lastKey = null;
-      while (rows.hasNext()) {
-        Row row = rows.next();
-        lastKey = row.getKey();
-        if (prefixes.stream().anyMatch(prefix -> row.getKey().startsWith(prefix))) {
-          bulkDelete.add(row.getKey(), Mutation.create().deleteRow());
-        }
+    Query query =
+        Query.create(tableId)
+            .filter(FILTERS.chain().filter(repoFilter(prefixes)).filter(FILTERS.value().strip()));
 
-        if (bulkDelete.getEntryCount() == MAX_BULK_MUTATIONS) {
-          dataClient.bulkMutateRows(bulkDelete);
-          bulkDelete = BulkMutation.create(tableId);
-        }
-      }
-      if (lastKey == null) {
-        break;
+    try (Batcher<RowMutationEntry, Void> batcher = dataClient.newBulkMutationBatcher(tableId)) {
+      ServerStream<Row> rows = dataClient.readRows(query);
+      for (Row row : rows) {
+        batcher.add(RowMutationEntry.create(row.getKey()).deleteRow());
       }
-      paginator.advance(lastKey);
-      rows = dataClient.readRows(nextQuery(paginator)).iterator();
-    }
-
-    if (bulkDelete.getEntryCount() > 0) {
-      dataClient.bulkMutateRows(bulkDelete);
+      batcher.sendOutstanding();
```

Comment:
Why is `sendOutstanding` needed? Doesn't `.close()` do that implicitly? Similar for the other changes.

Reply:
Indeed `close()` seems enough, but I was trying to stick as close as possible to the examples [here](https://cloud.google.com/bigtable/docs/delete-data#delete-stream-batch) and they use `.flush()` at the end of the iteration (which is the blocking version of `sendOutstanding()`.

I think we can remove all of them.